Adding configurable option for retry_limit on the AWS SDK, fixes #214

tyler-ball opened the pull request:
\cc @jkeiser @irvingpop @randomcamel 

The README and CHANGELOG should detail these changes - let me know if it does not.

Fixes #214

irvingpop commented on lib/chef/provisioning/aws_driver/driver.rb:
```diff
@@ -73,13 +73,15 @@ def initialize(driver_url, config)
         :http_proxy => credentials[:proxy_uri] || nil,
-        logger: Chef::Log.logger
+        logger: Chef::Log.logger,
+        retry_limit: Chef::Config.chef_provisioning[:aws_retry_limit] || 3
```
Hey @tyler-ball is there any reason the default retry limit shouldn't match the example in the readme?   25 seems like a sane number,  3 feels too low to be practical.

tyler-ball replied:
Sure, I can do 25.  Do you know what the default limits are from AWS before requests get throttled?

thommay commented:
Should you mention that the default max is 25 requests? because it's not clear to me from the docs that that's the case

tyler-ball commented:
Thanks @thommay - I added a note about the current default and dropped the default down to 5 because I think 25 is a big excessive.  I also updated the example so it shows how to set it for 10 retries instead of 25.

thommay commented:
:+1:
